### ISSUE TITLE
Add E2E tests with OpenSSL

### ISSUE THREAD

**Issue by Sean-Der** [pion/dtls#11]
Add some basic testing using s_client + s_server

The author should also refactor/re-use our existing E2E testing code so that we don't duplicate in both places.

**Comment by backkem** [comment]
We should probably implement these types of tests globally so we can throw them at any layer of the stack. The beginnings of this are [here](https://github.com/pions/webrtc/blob/dtls-native/internal/transport/test/test.go). Eventually it could look like [this](https://github.com/libp2p/go-libp2p-transport/blob/master/test/utils.go).

**Comment by igolaizola** [comment]
OpenSSL tests have been added in https://github.com/pion/dtls/pull/193

TODO:
- [ ] Make OpenSSL work with ed25519 test

**Comment by at-wat** [comment]
I found that our implementation of Ed25519 signature calculation and hash used in `ServerKeyExchange`/`CertificateVerify` messages are wrong. Since both of our server and client have same bug, they can communicate each other.
I have made a fix for it, and now I'm cleaning the code.

One bad news is that OpenSSL haven't support DTLS Ed25519 yet.
I could confirm my fix works with this openssl branch: https://github.com/openssl/openssl/pull/11115
